### PR TITLE
Post-huge update fix #2: deprecations

### LINKED ISSUE #2: Can't kill the echo bot example
First, great work! It has been very useful and fun to me.
I copy/pasted the example, but can't be interrupted with the KeyboardInterrupt signal on Gentoo.
IMHO one possible fix could be change the main loop at the end:
try:
    while 1:
        time.sleep(20)
except KeyboardInterrupt:
    tb.stop_polling()

### PR DESCRIPTION
**Pull request by Badiboy**
## Description
Deprecation fixes

### CODE REVIEW

**Comment by Badiboy** [comment]
(Not finished: more deprecation fixes are planned).

**Comment by Badiboy** [comment]
@coder2020official I'm done. 
.

> Improved documentation for the field message of the class [callbackQuery](https://core.telegram.org/bots/api#callbackquery) and the field pinned_message of the class [Message](https://core.telegram.org/bots/api#message) by adding the classes [MaybeInaccessibleMessage](https://core.telegram.org/bots/api#maybeinaccessiblemessage) and [InaccessibleMessage](https://core.telegram.org/bots/api#inaccessiblemessage).

Is also done.

**Comment by coder2020official** [comment]
image
Why extra spaces? If you really want a formatting style - use black

**Comment by coder2020official** [comment]
I have previously suggested such approach
https://github.com/eternnoir/pyTelegramBotAPI/issues/1853

This PR seems to be mostly about style fixes. It would be better to leave style fixes for another PR.

Your implementation of inaccessible message makes sense, that part I approve

**Comment by Badiboy** [comment]
> Why extra spaces

By mistake. It was occasionally formated, I did not paid attention.

**Comment by Badiboy** [comment]
> This PR seems to be mostly about style fixes.

It is not. I MAY fix some style only as a "supporting" activity.

**Comment by coder2020official** [comment]
Don't worry about style, spaces, etc - I will use black for the whole project if you approve

**Comment by coder2020official** [comment]
Can you also please obey the order for each parameter in description? I get the order from the api doc, it is easier to navigate and stuff

**Comment by coder2020official** [comment]
All of these were generated using api doc parsing. That is why most types have "ordered" parameters according to the doc

**Comment by Badiboy** [comment]
> Don't worry about style, spaces, etc - I will use black for the whole project if you approve

I still don't think we need total redesign.

**Comment by coder2020official** [comment]
I may update the doc strings in future when I find my old script or have time to recreate one

**Comment by Badiboy** [comment]
> It's better if you take a look at your changes. I have seen some in message class, didn't check all classes though

I checked and didn't find **any** reordering made by me. Please, point exactly if I'm wrong.

**Review by coder2020official — COMMENTED** [review]
This is all I could spot. And regarding the parameters order - sorry about that, I see nothing regarding order changed. 

Some questions:
- Did you make changes to async descriptions too? Would be nice if you could
- I see you are passing parameters by their name for init file. Did you check carefully so there are no typos?

That is it i guess, otherwise LGTM

**Comment by coder2020official** [comment]
Oh wait, my comments got removed.

File: telebot/types.py [hunk 1]
```diff
@@ -7703,7 +7788,7 @@ def de_json(cls, json_string):
         obj = cls.check_json(json_string)
         return cls(**obj)
 
-    def __init__(self, request_id: int, chat_id: int) -> None:
+    def __init__(self, request_id: int, chat_id: int, **kwargs) -> None:
```
Comment (coder2020official) [inline telebot/types.py:7791]: What is the purpose of kwargs when we don't use it?

File: telebot/types.py [hunk 2]
```diff
@@ -957,50 +954,50 @@ class Message(JsonDeserializable):
...
+        identifier. This number may have more than 32 significant bits and some programming languages may have
+        difficulty/silent defects in interpreting it. But it has at most 52 significant bits, so a signed 64-bit integer or
         double-precision float type are safe for storing this identifier.
     :type migrate_from_chat_id: :obj:`int`
 
-    :param pinned_message: Optional. Specified message was pinned. Note that the Message object in this field will not 
+    :param pinned_message: Optional. Specified message was pinned. Note that the Message object in this field will not
         contain further reply_to_message fields even if it is itself a reply.
-    :type pinned_message: :class:`telebot.types.Message`
+    :type message: :class:`telebot.types.Message` or :class:`telebot.types.InaccessibleMessage`
```
Comment (coder2020official) [inline telebot/types.py:985]: Supposed to be pinned_message for type
```suggestion
    :type pinned_message: :class:`telebot.types.Message` or :class:`telebot.types.InaccessibleMessage`
```

File: telebot/types.py [hunk 3]
```diff
@@ -8050,7 +8137,7 @@ class ReactionTypeCustomEmoji(ReactionType):
     :rtype: :class:`ReactionTypeCustomEmoji`
     """
 
-    def __init__(self, custom_emoji: str) -> None:
+    def __init__(self, custom_emoji: str, **kwargs) -> None:
```
Comment (coder2020official) [inline telebot/types.py:8140]: btw, custom_emoji should be turned to custom_emoji_id:
https://core.telegram.org/bots/api#reactiontypecustomemoji

**Review by coder2020official — CHANGES_REQUESTED** [review]

**Comment by Badiboy** [comment]
> * Did you make changes to async descriptions too? Would be nice if you could

For this update - no. There are no bugfixes, so these updates are not critical. If you like - you can bring them to Async or not, it's up to you as the Async version designer. Currently I'm not able to spend more days on Async: I overtake the available time to effort on this update. :(((

> * I see you are passing parameters by their name for init file. Did you check carefully so there are no typos?

The purpose of pass all named parameters as named is to avoid mistakes on updates related to sequence of parameters. With named values calls are now safe on parameters order misses.

I did this update as accurate as I can compare everything. As far as tests are passing - syntactically these calls are correct. Of course, there MAY BE some mistakes, I'll fix them on occurence.

File: telebot/types.py (hunk 1 above)
Comment (Badiboy) [reply telebot/types.py:7791]: > What is the purpose of kwargs when we don't use it?

IMO **kwargs saves library from updates of API. If there appears new fields - they will be autoparsed and passed to init. If there will be no dummy **kwargs to accept them - init call will fail with incorrect number of aguments. Isn't it?

We (you :) ) created these **kwargs long ago when invented this new method of calling init: I just allign some methods to the same signature as others.

File: telebot/types.py (hunk 3 above)
Comment (Badiboy) [reply telebot/types.py:8140]: Thanks, fixed.

**Comment by coder2020official** [comment]
I think that is it

**Comment by Badiboy** [comment]
Thank you for review!

**Comment by Badiboy** [comment]
@coder2020official 
1. What about version? 4.15.0 or 5.0.0? :) 
2. Do we have to make anything else? Or I can start testing update on my systems?

**Comment by coder2020official** [comment]
I wouldn't rush tbh. I am planning to:
- Update docstrings
- Update docs
- And try out black formatting

Anyways, I think we have to wait for some time. We don't want surprises, do we? And regarding version number - we need something more than just an API update for version 5. So I guess it would be better to stick with 4.15.0

**Comment by Badiboy** [comment]
> Anyways, I think we have to wait for some time. We don't want surprises, do we?

Tests are part of activities avoiding surprises ))

**Comment by coder2020official** [comment]
The project is not fully covered by tests which makes it harder for us to rely on them

**Pull request merged**